src/functions/operators.js

```diff
@@ -30,36 +31,60 @@ defineFunction({
             let mode = "";
 
             // Consolidate Greek letter function names into symbol characters.
-            const temp = html.buildExpression(group.value.value, options, true);
+            const temp = html.buildExpression(
+                group.value.value, options.withFont("mathrm"), true);
 
             // All we want from temp are the letters. With them, we'll
             // create a text operator similar to \tan or \cos.
             for (let i = 0; i < temp.length; i++) {
-                letter = temp[i].value;
+                const child = temp[i];
 
-                // In the amsopn package, \newmcodes@ changes four
-                // characters, *-/:’, from math operators back into text.
-                // Given what is in temp, we have to address two of them.
-                letter = letter.replace(/\u2212/, "-");   // minus => hyphen
-                letter = letter.replace(/\u2217/, "*");
+                if (child instanceof domTree.symbolNode) {
+                    letter = child.value;
 
-                // Use math mode for Greek letters
-                mode = (/[\u0391-\u03D7]/.test(letter) ? "math" : "text");
-                output.push(buildCommon.mathsym(letter, mode));
+                    // In the amsopn package, \newmcodes@ changes four
+                    // characters, *-/:’, from math operators back into text.
+                    // Given what is in temp, we have to address two of them.
+                    letter = letter.replace(/\u2212/, "-");   // minus => hyphen
+                    letter = letter.replace(/\u2217/, "*");
+
+                    // Use math mode for Greek letters
+                    mode = (/[\u0391-\u03D7]/.test(letter) ? "math" : "text");
+                    output.push(buildCommon.mathsym(letter, mode));
+                } else {
+                    output.push(child);
+                }
             }
         }
-        return buildCommon.makeSpan(["mop"], output, options);
+        const span = buildCommon.makeSpan(["mop"], output, options);
+        return span;
     },
 
     mathmlBuilder: (group, options) => {
         // The steps taken here are similar to the html version.
         let output = [];
         if (group.value.value.length > 0) {
-            const temp = mml.buildExpression(group.value.value, options);
+            const temp = mml.buildExpression(
+                group.value.value, options.withFont("mathrm"));
 
             let word = "";
             for (let i = 0; i < temp.length; i++) {
-                word += temp[i].children[0].text;
+                if (utils.contains(["mi", "mn", "mo"], temp[i].type)) {
```

Comment:
1. This is not backward-compatible.
2. It's not obvious to me that these are the only allowable possibilities inside an `operatorname`, and not, say, `menclose` `mrow` or `mpadded` or any of the stuff on the left sidebar [here](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mpadded). I see examples of sub/superscripts within `operatorname` [here](https://tex.stackexchange.com/a/100064) and [here](https://tex.stackexchange.com/q/67949) among others.

Reply:
Good point.  I'll try to make this more general.